### PR TITLE
[WIP] added server to set base_frame, included some wait for transform and error handling

### LINKED ISSUE #64: handle canTransform in cob_linear_nav properly
throws a lot of exceptions and warnings on the terminal

### PR DESCRIPTION
**Pull request by stefandoerr**
fixes #64

### CODE REVIEW

**Comment by floweisshardt** [comment]
Looks good, I'll test this with docking later today

File: cob_linear_nav/src/cob_linear_nav.cpp [hunk 1]
```diff
@@ -435,7 +465,10 @@ void NodeClass::publishVelocitiesGlobal(double vx, double vy, double theta) {
   cmd_global.header.frame_id = global_frame_;
   cmd_global.vector.x = vx;
   cmd_global.vector.y = vy;
-  try { tf_listener_.transformVector(robot_frame_, cmd_global, cmd_robot);
+  try
+  {
+    tf_listener_.waitForTransform(robot_frame_, cmd_global.header.frame_id, ros::Time(0), ros::Duration(0.02));
+    tf_listener_.transformVector(robot_frame_, cmd_global, cmd_robot);
```
Comment (benmaidel) [inline cob_linear_nav/src/cob_linear_nav.cpp:462]: Can this be possible? `transformVector` has no transform time at all...
normally you should set a timestamp inside the header!?
@ipa-mig

File: cob_linear_nav/src/cob_linear_nav.cpp [hunk 2]
```diff
@@ -321,11 +329,33 @@ class NodeClass
...
+    }
+    catch(tf::TransformException& ex){ROS_ERROR("%s",ex.what());}
 
     vec_stamped.vector = odometry->twist.twist.angular;
     vec_stamped.header.frame_id =  "base_footprint";
-    tf_listener_.transformVector(robot_frame_, vec_stamped, robot_twist_angular_robot_);
+    try
+    {
+      tf_listener_.waitForTransform(robot_frame_, vec_stamped.header.frame_id, ros::Time(0), ros::Duration(0.01));
+      tf_listener_.transformVector(robot_frame_, vec_stamped, robot_twist_angular_robot_);
```
Comment (benmaidel) [inline cob_linear_nav/src/cob_linear_nav.cpp:332]: Can this be possible? `transformVector` has no transform time at all...
normally you should set a timestamp inside the header!?
@ipa-mig

File: cob_linear_nav/src/cob_linear_nav.cpp [hunk 3]
```diff
@@ -321,11 +329,33 @@ class NodeClass
 
     vec_stamped.vector = odometry->twist.twist.linear;
     vec_stamped.header.frame_id =  "base_footprint";
-    tf_listener_.transformVector(robot_frame_, vec_stamped, robot_twist_linear_robot_);
+    try
+    {
+      tf_listener_.waitForTransform(robot_frame_, vec_stamped.header.frame_id, ros::Time(0), ros::Duration(0.01));
+      tf_listener_.transformVector(robot_frame_, vec_stamped, robot_twist_linear_robot_);
```
Comment (benmaidel) [inline cob_linear_nav/src/cob_linear_nav.cpp:323]: Can this be possible? `transformVector` has no transform time at all...
normally you should set a timestamp inside the header!?
@ipa-mig

File: cob_linear_nav/src/cob_linear_nav.cpp (hunk 2 above)
Comment (mgruhler) [inline cob_linear_nav/src/cob_linear_nav.cpp:332]: @ipa-bnm http://docs.ros.org/indigo/api/tf/html/c++/classtf_1_1TransformListener.html

it takes the stamp from the header of the Stamped Type...

Comment (stefandoerr) [inline cob_linear_nav/src/cob_linear_nav.cpp:332]: @ipa-mig the stamp of the header was never set!?

**Comment by benmaidel** [comment]
CR: ok

File: cob_linear_nav/src/cob_linear_nav.cpp (hunk 2 above)
Comment (mgruhler) [inline cob_linear_nav/src/cob_linear_nav.cpp:332]: whoops. you are right. How could this ever have worked?

Comment (benmaidel) [inline cob_linear_nav/src/cob_linear_nav.cpp:332]: :)

**Comment by mgruhler** [comment]
CR: ok

@ipa-fwm fine from your side as well?

**Comment by floweisshardt** [comment]
FR:OK

**Pull request merged**